tools/ci-scripts/check-aws-config

```diff
@@ -42,11 +42,11 @@ cargo hack check --feature-powerset --exclude-all-features --exclude-features al
 
 echo "${C_YELLOW}## Testing each feature in isolation${C_RESET}"
 # these features are missed by the following check because they are grouped
-cargo hack test --each-feature --include-features rt-tokio,client-hyper,rustls --exclude-no-default-features
+cargo hack test --each-feature --include-features rt-tokio,client-hyper,rustls,default-http-connector --exclude-no-default-features
 
 # This check will check other individual features and no-default-features
 # grouping these features because they don't interact
-cargo hack test --feature-powerset --exclude-all-features --exclude-features allow-compilation --group-features rt-tokio,client-hyper,rustls
+cargo hack test --feature-powerset --exclude-all-features --exclude-features allow-compilation --group-features rt-tokio,client-hyper,rustls,default-http-connector
```

Comment:
Ah good, `default-http-connector` does not increase the number of combinations, as it can be part of the existing feature group.

Reply:
We were in fact timing out as you suspected prior to this 😆

Reply:
I've reduced the feature powerset for `aws-smithy-runtime` as well because it was timing out after 30 mins: https://github.com/smithy-lang/smithy-rs/pull/3939/commits/c50a3d2f06191f9430b4a37c503cd7c49ebebf11

Let me know if you see anything wrong with this but my thought process is we have several features that exist to keep things working for a legacy ecosystem we are moving from. Most of these flags are now controlling re-exports. Not testing them combinatorially is probably good/ok going forward.